### PR TITLE
De-dupe readme and docs site

### PR DESCRIPTION
**Pull request by jthegedus**
# Summary

This is the last major part of the work done in #422 and #440 

As discussed in #422 this change will separate out the **WHAT & WHY** from the **HOW**. The **proposed** readme is simply information for newcomers about asdf-vm and why they would use it, redirecting to the documentation site for actual instructions.

It would be good if all @asdf-vm/core were aware of this change before it merged so we are on the same page going forward about additions to the readme.

### The new readme

I've already used the proposed readme as content on the home page of the docs site. Here is what it would look like on GitHub https://github.com/asdf-vm/asdf/blob/master/docs/NEW_README.md Here it is on the site currently: https://asdf-vm.github.io/asdf/#/?id=why-use-asdf-vm

### why no quickstart in the proposed readme?
I had considered having a "quickstart" section in the new readme, but unlike a simple npm package, it's not a single line setup, and the doc site was made specifically so people could select their terminal system and get the exact information, no more, no less. Because of this, the **readme is sparse, clear and direct**, similar to the [prettier](https://github.com/prettier/prettier#opinionated-code-formatter) or [docsify](https://github.com/docsifyjs/docsify/) readme's, which are successful and popular tools with no complaints about the lack of quickstart in the readmes.

### the prev readme
As there are still some WIP changes to the main Readme, it has been "backed-up" into the `docs/` folder, and can be removed once a re-sync with the `docs/` content has occurred (I will attempt that ASAP after merging this).

### Other Information

* This should go a fair way to resolving #438 which was one of the reasons I started contributing to asdf-vm in the first place. 

### Link to the doc site from repo header

I am unable to edit the repo topics, or the repo website link in the repo header so  could you, @HashNuke add https://asdf-vm.github.io/asdf/#/ to that. It would be greatly appreciated.

### CODE REVIEW

**Comment by jthegedus** [comment]
@Stratus3D Any thoughts on this? I would like to get the okay from most the maintainers before making such a drastic change, and I would like to make it soon before further PRs to the existing readme

File: README.md [hunk 1]
```diff
@@ -1,374 +1,29 @@
...
-[plugins]: https://github.com/asdf-vm/asdf-plugins
-[plugin_api]: https://github.com/asdf-vm/asdf/blob/master/docs/plugins-create.md
 
-## INSTALLATION
+## Why use asdf-vm?
 
-If you're on macOS you can [Install using Homebrew](https://github.com/asdf-vm/asdf#homebrew-on-macos)
+- single CLI for multiple languages
+- consistent commands to manage all your languages
+- single `.tool-versions` config file per project
```
Comment (Stratus3D) [inline README.md]: Might also be nice to mention the support for other version files like `.ruby-version`, `.node-version`, etc...

File: README.md [hunk 2]
```diff
@@ -1,374 +1,29 @@
-# asdf [![Build Status](https://travis-ci.org/asdf-vm/asdf.svg?branch=master)](https://travis-ci.org/asdf-vm/asdf)
+# asdf-vm [![Build Status](https://travis-ci.org/asdf-vm/asdf.svg?branch=master)](https://travis-ci.org/asdf-vm/asdf)
 
-### _extendable version manager_
```
Comment (Stratus3D) [inline README.md]: I think we could keep both these lines if we wanted to.

```
### _extendable version manager_

**Manage multiple runtimes with a single CLI tool**
```

**Comment by Stratus3D** [comment]
@jthegedus and thanks for keeping the old readme checked into the repo, I'm sure it will come in handy when we are reviewing future documentation PRs and issues.

**Comment by jthegedus** [comment]
@Stratus3D I'll add your suggestions, then are we happy enough for me to merge this today?

**Comment by jthegedus** [comment]
I agree with the feedback is all 😄 Awesome, I'll get this in today and
then work on migrating existing readme PRs and resync any readme changes
that occurred while making this. Nearly there!

**Pull request merged**